apps/rush-lib/src/utilities/RushCiMode.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as isCI from 'is-ci';
+import * as colors from 'colors';
+
+/**
+ * Determines if Rush is running in CI mode or NON-CI mode
+ */
+export class RushCiMode {
+
+  private static _rushCiMode: RushCiMode;
+
+  private readonly _forceCI: boolean;
+  private readonly _forceNONCI: boolean;
+
+  private static _initialized: boolean;
+
+  private constructor(overrideMode: string | undefined) {
+    this._forceCI = false;
+    this._forceNONCI = false;
+    if (!overrideMode) {
+      // Nothing to do.
+      return;
+    }
+
+    if (overrideMode === 'CI') {
```

Comment:
These logging messages should print something about why it's running in whatever mode it's running in, and about how to override.